### PR TITLE
Syslog rewrite

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28637

## Description
Syslog v2 rewrite from server integration.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: Some of the filters for syslog that existed in V1 were not used. as can be seen in the issue above of the rewrite, it was decided to only implement message regex filter at first, and enhance with more filtering if requested.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tomneeman151293** [comment]
Validations are failing on:
- Missing TPB - the integration only implements long-running-execution so I didnt implement one
- using dev docker image - after this PR will be approved, i will merge my dockerfile PR https://github.com/demisto/dockerfiles/pull/6358, and those validations will pass

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:316]: when exactly is this happening? let's document and briefly explain the sampling logic

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:277]: nice impl, but better to provide the field description in the values of the dict, or which is used in which format. you can do it hard-coded.

**Review by dantavori — CHANGES_REQUESTED** [review]
🥇

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md:7]: yeah, shouldn't be any issue, as the only problem is where the long-running needs to be configured on a single IP (in load balancing the IP might change so the server might not listen to the port in the expected IP

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml:22]: https://stackoverflow.com/questions/69985784/syslog-rfc5424-vs-rfc6587

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.yml:44]: leaving it as dev until dockerfiles pr is merged

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2.py
Comment (tomneeman151293) [reply Packs/Syslog/Integrations/Syslogv2/Syslogv2.py:287]: i will delete

File: Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md
Comment (dantavori) [inline Packs/Syslog/Integrations/Syslogv2/Syslogv2_description.md:3]:
```suggestion
```

**Review by dantavori — COMMENTED** [review]
will approve after the demo.
try to think how we can test it in a TPB

**Comment by tomneeman151293** [comment]
@dantavori Updated docker image to prod image

**Review by dantavori — APPROVED** [review]
looks good 👍 please open an issue about the test playbook as we discussed

**Comment by tomneeman151293** [comment]
@dantavori  https://github.com/demisto/etc/issues/44245

**Comment by tomneeman151293** [comment]
@Itay4 need to force merge as we dont have a TPB for this integration.

**Comment by tomneeman151293** [comment]
@Itay4 Force merge is required. Validation is failing on missing TPB. A separate issue was created to find a dedicated instance to syslog TPB to be developed in the future

demisto/etc#44245

**Comment by Itay4** [comment]
merging due to the above

**Pull request merged**
